components/earnings/commons/earnings.lua

```diff
@@ -0,0 +1,132 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:Earnings
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Earnings = {}
+local MathUtils = require('Module:Math')
+local String = require('Module:StringUtils')
+local Class = require('Module:Class')
+
+---
+-- Entry point for players and individuals
+-- @player - the player/individual for whom the earnings shall be calculated
+-- @year - (optional) the year to calculate earnings for
+-- @mode - (optional) the mode to calculate earnings for
+function Earnings.calculateForPlayer(args)
+	local player = args.player
+
+    if String.isEmpty(player) then
+        return 0
+    end
+	player = mw.ext.TeamLiquidIntegration.resolve_redirect(player)
+
+    local money = 0
+    local condition = '([[players_p1::' .. player .. ']] OR'
+        .. '[[players_p2::' .. player .. ']] OR'
+        .. '[[players_p3::' .. player .. ']] OR'
+        .. '[[players_p4::' .. player .. ']] OR'
+        .. '[[players_p5::' .. player .. ']] OR'
+        .. '[[players_p6::' .. player .. ']] OR'
+        .. '[[players_p7::' .. player .. ']] OR'
+        .. '[[players_p8::' .. player .. ']] OR'
+        .. '[[players_p9::' .. player .. ']] OR'
+        .. '[[players_p10::' .. player .. ']] OR'
+        .. '[[participant::' .. player .. ']])'
```

Comment:
Could add the possibility for custom prefix.
Some wikis still use `player` instead of `p`. It could also be useful if for instance a wiki want to grab the earnings for coach by some metric (`c` as prefix). Default prefix should be `p`.
Number of players needs to be customizable.